CONTRIBUTING.md

```diff
@@ -351,6 +351,226 @@ The third schema is `third.schema.json`, it extends both `second`, and transitiv
+
+XDM defines a schema descriptor of type `xdm:descriptorUpdatePolicy`, which describes several common methods of handling an update:
+
+* `xdm:updateMerge`: the data in the new object should be merged into the existing object; the method by which a merge is applied is defined by the application
+* `xdm:updateReplace`: the new data object should replace the existing data object
+* `xdm:updateTimeSeries`: the data is time series data, and the new object should be logged/collected without changing any existing data
+
+Update policies are defined using the UpdatePolicyDescriptor schema.
+
+#### Other Supported Schema Descriptors
+
+A number of additional schema descriptors are defined by XDM:
+
+* `xdm:descriptorIdentity`: allows a property in a schema to be used as an identity, even if it does not conform to the Identity schema.
+* `xdm:descriptorPrimaryKey`: allows a property other than `@id` to be flagged as the primary key for a schema
+* `xdm:descriptorInstantiable`: allows a schema to be flagged as 'instantiable', which may be used to differentiate schemas that define primary business objects versus supporting schemas intended to be embedded in another schema.
+
+#### Schema Descriptor Examples
+
+##### Example Relationship Descriptor
+
+We have two schemas, which form a parent/child relationship. The first is parent.json:
+
+```json
+{
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "$id": "https://ns.adobe.com/xdm/example/parent",
+  "title": "Parent",
+  "type": "object",
+  "meta:descriptors": {
+    "@type": "xdm:descriptorOneToMany",
+    "xdm:sourcePropery": "@id",
+    "xdm:destSchema": "https://ns.adobe.com/xdm/example/child",
+    "xdm:destProperty": "xdm:parent"
+  },
+  "properties": {
+    "@id": { "type": "string" }
+  }
+}
+```
+
+The second is child.json:
+
+```json
+{
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "$id": "https://ns.adobe.com/xdm/example/child",
+  "title": "Child",
+  "type": "object",
+  "properties": {
+    "@id": { "type": "string" },
+    "xdm:parent": { "type": "string" }
+  }
+}
+```
+
+The source schema in this example is Parent, which contains a single relationship descriptor describing a one-to-many relationship between objects of schema Parent to objects of schema Child.
+
+Note that the descriptor does not contain an '@id' or 'xdm:sourceSchema property', which are optional when the descriptor is embedded directly in the schema. If the relationship descriptor were to be written externally, it would look like this:
+
+```json
+{
+  "@id": "https://example.com/descriptors/1",
+  "@type": "xdm:descriptorOneToMany",
+  "xdm:sourceSchema": "https://ns.adobe.com/xdm/example/parent",
+  "xdm:sourcePropery": "@id",
+  "xdm:destSchema": "https://ns.adobe.com/xdm/example/child",
+  "xdm:destProperty": "xdm:parent"
+}
+```
```

Comment:
This example shows a classical one-to-many relationship, where an instance of "many" side has a property (foreign key) with a value being a primary key of an instance of "one" side. We need to differentiate it with the case where want to have on "one" side an array of references to "many" side. Should we have two separate cases for "oneToMany" and "manyToOne"?

Reply:
Hi @dmitry-shkolnik, I'm not sure I completely follow.  Are you describing a case where 1:M and "M:1" aren't really just reciprocal,  but need to have distinct semantics?  Can you give a specific example of that?

Reply:
I am talking about the case, where it is more convenient to have an array of pointers to the children in the parent than pointer to parent in each children. In that case, when the relationship  between two instances is created, we'd like to add @id of a child to array in parent, instead of setting `parent` property in the child. 

This is similar to aggregation (when an object has an array of children), but if we decide to store children separately, we cannot express that with the descriptors.

As an examples: each instance of the offer schema (which we are working on) has an array of representations (how the offer may look like on different channels /  placements). Right now representations are fully embedded into offer. But if we decide to make them a separate first-class citizens with their own id, we''ll have to get rid of array in offer and add parent property to the representation, which is an old-fashion rdbms way of modeling.

Reply:
Yes, that is definitely a case we haven't considered.  In general, the semantics of the relationship in that case don't change: you still have a 1:M relationship from parent to child.  But the relationship descriptor (as proposed) can't be applied to the denormalized form in your example.  I agree it would be good to be able to handle that, I'll need to think about how to approach it.